activerecord/lib/active_record/migration/compatibility.rb

```diff
@@ -104,11 +104,28 @@ def index_name_for_remove(table_name, options = {})
 
       class V5_0 < V5_1
         def create_table(table_name, options = {})
-          if ActiveRecord::Base.connection.adapter_name == "PostgreSQL"
+          connection_name = self.connection.adapter_name
+          if connection_name == "PostgreSQL"
             if options[:id] == :uuid && !options[:default]
               options[:default] = "uuid_generate_v4()"
             end
           end
+
+          # Since 5.1 Postgres adapter uses bigserial type for primary
+          # keys by default and MySQL uses bigint. This compat layer makes old migrations utilize
+          # serial/int type instead -- the way it used to work before 5.1.
```

Comment:
That's where I was thinking we could define defaults that must work well, and teach our own adapters to support them.

I'm fine with forcing changes onto adapters\* -- I just don't want them to have to reach outside their classes to patch this `V5_0` class, for example.

---

\* Ultimately we'd like to get to a point where we don't do that so much, but right now, it's par for the course. So while big-picture unfortunate, it's not a negative against a particular approach here & now.